contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -742,51 +768,152 @@ have a built-in `nil` value.
 
 ## Defaulting
 
-Default resource values are API version-specific, and they are applied during
-the conversion from API-versioned declarative configuration to internal objects
-representing the desired state (`Spec`) of the resource. Subsequent GETs of the
-resource will include the default values explicitly.
-
-Incorporating the default values into the `Spec` ensures that `Spec` depicts the
-full desired state so that it is easier for the system to determine how to
-achieve the state, and for the user to know what to anticipate.
+In general we want default values to be explicitly represented in our APIs,
```

Comment:
Might be a can of worms, but do we want to mention this corner case of defaulting fields in pod spec? I don't think we ever came to a conclusion on https://github.com/kubernetes/community/pull/4571

Reply:
mm... I view the current state of pod defaulting a bug we're ignoring/working around, not a convention we should document here

Reply:
Ditto.  It's a problem we ought to not have.